src/re_bytes.rs

```diff
@@ -757,14 +757,19 @@ pub struct SplitN<'r, 't> {
 impl<'r, 't> Iterator for SplitN<'r, 't> {
     type Item = &'t [u8];
 
-    fn next(&mut self) -> Option<&'t [u8]> {
-        if self.n == 0 {
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.n <= 0 {
```

Comment:
How can a `usize` ever drop below 0?

Reply:
Keep calling `.next()` even after it's returned `None`, and n goes past 0 to negative...

Reply:
I'm pretty sure it won't because this if block will always fire and cause a return before n gets decremented. Also, because n is a usize there is no pattern of bits which constitutes a negative number.

Reply:
Yes ofc you're right, thanks. Updated